### PR TITLE
Improved diff'ing of JAR packages and Java classes

### PR DESCRIPTION
**Pull request by tlil**

### CODE REVIEW

**Comment by tlil** [comment]
Ignore - was meaning to raise this against `tlil:master`

File: modules/Internals/Tools/java-dump.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+function decompile {
```
Comment (lvc) [inline modules/Internals/Tools/java-dump.sh]: Hi,

Thank you for the patch.

Why not just adding '-s -c -private -verbose' to javap call in the `showFile` function?

File: pkgdiff.pl [hunk 2]
```diff
@@ -3232,12 +3253,21 @@ ($$$$$)
     <title>
         $Title
     </title>
+    <link rel=\"stylesheet\" type=\"text/css\" href=\"https://cdn.datatables.net/1.10.16/css/jquery.dataTables.min.css\" />
```
Comment (lvc) [inline pkgdiff.pl]: Importing of files from web can be accepted only as an option.

Comment (tlil) [reply pkgdiff.pl]: Hi @lvc - I've actually updated this branch (as can be seen here: https://github.com/corda/pkgdiff/pull/1). Please ignore this one. I will raise a new PR against lvc/pkgdiff if you're interested in incorporating the changes into the main repo.

File: modules/Internals/Tools/java-dump.sh (hunk 1 above)
Comment (tlil) [reply modules/Internals/Tools/java-dump.sh]: I used the `-c` flag at first, but we're running a diff against the full version of OpenJDK's `rt.jar`, so this utility provides a more condensed summary. I'll re-raise a PR and include two alternative options for this; summarised and full method bodies. The former, which is what I'm using here, essentially strips any numeric references from the bytecode and runs SHA256 on the set of instructions that make up each method body.

File: pkgdiff.pl (hunk 2 above)
Comment (tlil) [reply pkgdiff.pl]: @lvc I've raised https://github.com/lvc/pkgdiff/pull/31 instead

File: modules/Internals/Tools/java-dump.sh (hunk 1 above)
Comment (lvc) [reply modules/Internals/Tools/java-dump.sh]: FYI about OpenJDK testing:

This page may be of interest to you: https://abi-laboratory.pro/java/tracker/timeline/openjdk/

Thank you.

Comment (tlil) [reply modules/Internals/Tools/java-dump.sh]: Yep, I saw that one as well. But thanks for pointing me to it. :+1:

**Pull request closed without merging**